services/py-api/src/server/handlers/participants_handlers.py

```diff
@@ -21,14 +25,27 @@ def __init__(self, service: ParticipantRegistrationService) -> None:
         self._service = service
 
     async def create_participant(
-        self, response: Response, input_data: ParticipantRequestBody
+        self, response: Response, input_data: ParticipantRequestBody, jwt_token: Optional[str] = None,
     ) -> ParticipantRegisteredResponse | ErrResponse:
+        
         if input_data.is_admin and input_data.team_name:
             result = await self._service.register_admin_participant(input_data)
 
-        # TODO: Implement this when the invite participant case is done
-        # elif input_data.is_admin is False and input_data.team_name:
-        #     result = await self._service.register_invite_participant(input_data)
+        elif input_data.is_admin is False and input_data.team_name:
+            # Decode the token
+            decoded_result = JwtUtility.decode_data(token=jwt_token, schema=JwtUserData)
+            if is_err(decoded_result):
+                response.status_code = status.HTTP_401_UNAUTHORIZED
+                return ErrResponse(error=f"JWT decoding failed: {decoded_result.err_value}")
```

Comment:
you can just print error=decoded_result.err_value

decoded_result.err_value could be: 
1. "The decoded token does not correspond with the provided schema."
2. "The JWT token has expired."
3. "The JWT token has invalid signature."

So these messages are sufficient on their own

Reply:
Ok but since I moved this logic to the service layer I should return the Err not an Error Response right? Then the ErrorResponse should be in the handler? If so should I create new exceptions for these messages so we can use the same isinstance pattern?